src/log.rs

```diff
@@ -41,6 +41,17 @@ macro_rules! error {
     }};
 }
 
+#[macro_export]
+macro_rules! error_network {
```

Comment:
The macro is used only once currently, makes sense to replace other instances of `emit_event!` with `ErrorNetwork` in this PR?

Reply:
i'd do that in a subsequent pr (in fact, i tried to do it without a macro but i failed, the easiest thing was to copy+paste the error! macro :)